src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useState } from 'react';
+import { Todo } from '../../Types/Todo';
+import usersFromServe from '../../api/users';
+import { getUserById } from '../../services/user';
+import { getNewTodoId } from '../../services/id';
+
+type Props = {
+  onAdd: (todo: Todo) => void;
+  todos: Todo[];
+};
+
+export const TodoForm: React.FC<Props> = ({ onAdd, todos }) => {
+  const [userId, setUserId] = useState(0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const resetForm = () => {
+    setTitle('');
+    setUserId(0);
+    setHasUserIdError(false);
+    setHasTitleError(false);
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const handleUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserIdError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserIdError(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    onAdd({
+      id: getNewTodoId(todos),
+      title,
+      userId,
+      completed: false,
+      user: getUserById(userId),
+    });
+
+    resetForm();
+  };
+
+  return (
+    <form action="/api/todos" method="POST" onSubmit={handleSubmit}>
+      <div className="field">
+        <label className="lable" htmlFor="todo-title">
+          Title:&nbsp;
+        </label>
+        <input
+          type="text"
+          data-cy="titleInput"
+          placeholder="Enter a title"
+          value={title}
+          onChange={handleTitleChange}
+        />
+        {hasTitleError && <span className="error">Please enter a title</span>}
+      </div>
+
+      <div className="field">
+        <label className="lable" htmlFor="user-title">
+          User:&nbsp;
+        </label>
+        <select
+          data-cy="userSelect"
+          value={userId}
+          onChange={handleUserIdChange}
+        >
+          <option value="0" disabled>
+            Choose a user
+          </option>
+          {usersFromServe.map(user => (
+            <option value={user.id} key={user.id}>
+              {user.name}
+            </option>
+          ))}
```

Comment:
The list of users should be passed as a prop from the `App` component rather than being imported directly.

Reply:
> The list of users should be passed as a prop from the `App` component rather than being imported directly.

Good point